src/sentry/static/sentry/app/components/errorBoundary.tsx

```diff
@@ -52,38 +65,40 @@ class ErrorBoundary extends React.Component {
   }
 
   render() {
-    if (this.state.error) {
-      const {customComponent, mini, message, className} = this.props;
+    const {error} = this.state;
+
+    if (!error) {
+      //when there's not an error, render children untouched
+      return this.props.children;
+    }
 
-      if (customComponent) {
-        return customComponent;
-      }
+    const {customComponent, mini, message, className} = this.props;
 
-      if (mini) {
-        return (
-          <Alert type="error" icon="icon-circle-exclamation" className={className}>
-            {message || t('There was a problem rendering this component')}
-          </Alert>
-        );
-      }
+    if (customComponent) {
+      return customComponent;
+    }
 
+    if (mini) {
       return (
-        <Wrapper>
-          <DetailedError
-            heading={getExclamation()}
-            message={t(
-              `Something went horribly wrong rendering this page.
-We use a decent error reporting service so this will probably be fixed soon. Unless our error reporting service is also broken. That would be awkward.
-Anyway, we apologize for the inconvenience.`
-            )}
-          />
-          <StackTrace>{this.state.error.toString()}</StackTrace>
-        </Wrapper>
+        <Alert type="error" icon="icon-circle-exclamation" className={className}>
+          {message || t('There was a problem rendering this component')}
+        </Alert>
       );
-    } else {
-      //when there's not an error, render children untouched
-      return this.props.children;
```

Comment:
Nice change. There's a eslint rule that prevents this pattern of 
```
if (condition) {
  return;
} else {
  return;
}
```
in favor of 
```
if (condition) {
  return;
} 

return;
```

Is it part of our rules? If not, I'll find it and propose that it'll be added.

Reply:
That would be great!